recipes/automated-testing.md

```diff
@@ -86,10 +86,26 @@ changes, we can make minimal changes to the test suite and/or mocked data.
 
 ### Integration Tests
 
-With these tests, we test the application API endpoints and assert that they are
-actually working as expected.
+With these tests, we test the combined behavior of multiple components within the system.
 
-**TODO**: do we want to add that we should run full backend for these type of tests?
+#### Infrastructure
+
+Running the tests on test infrastructure should be preferred to mocking, unlike in unit tests. Ideally, a full application instance would be run, to mimic real application behavior as close as possible.
+This usually includes running the application connected to a test database, inserting fake data into it during the test setup, and doing assertions on the current state of the database. This also means integration test code should have full access to the test infrastructure for querying. 
+> [!NOTE]
+> Regardless of whether using raw queries or the ORM, simple queries should be used to avoid introducing business logic within tests.
+
+However, mocking can still be used when needed, for example when expecting side-effects that call third party services.
+
+#### Entry points
+
+Integration test entry points can vary depending on the application use cases. These include services, controllers, or the API. These are not set in stone and should be taken into account when making a decision. For example:
+- A use case that can be invoked through multiple different protocols can be tested separately from them, to avoid duplication. A tradeoff in this case is the need to write some basic tests for each of the protocols.
+- A use case that will always be invokeable through a single protocol might benefit enough from only being tested using that protocol. E.g. a HTTP API route test might eliminate the need for a lower level, controller/service level test. This would also enable the testing of authorization within the same test, which might not have been possible otherwise depending on the technology used.
+
+Multiple approaches can be used within the same application depending on the requirements, to provide sufficient coverage.
```

Comment:
Do we need to define a "use-case" to ensure different readers don't come up with different understandings?

Reply:
Hm, to be honest none comes to mind that wouldn't fit here. Do you have anything specific in mind?

Reply:
I think for a junior, "use-case" can mean method invocation, while in software architecture, use-case usually relates to business logic use-case encapsulated in an app service method, command or query (in CQRS).

Reply:
Never heard of it being used that way, maybe because it was explained as a business logic use case even in college in my case, which might not be the case for everyone. 
Do you think rewording "application use cases" to "business logic use cases" is enough, or you have something else in mind?

Reply:
Added to glossary, backlinks seemed like overkill for now